docs/development/pigs/pig-008.rst

```diff
@@ -0,0 +1,594 @@
+`SpectrumDataset`, `SpectrumDatasetOnOff` and `MapDatasetOnOff`. The on-off
+dataset require a background model for this to work.
+
+Unbinned simulation (sampling of event lists) is addressed in a separate PIG.
+
+
+`Datasets`
+----------
+
+To combine multiple datasets in a joint-likelihood analysis we propose to introduce
+a `Datasets` class. Its responsibility is to add up the log-likelihood values of
+the individual datasets and join the parameter lists. The `Datasets` class
+also represents the main interface to the `Fit` class.
+
+The following example shows how a joint-likelihood analysis of multiple observations
+is supposed to work:
+
+.. code ::
+
+    from gammapy.utils.fitting import Datatsets
+
+    model = SourceModel.read()
+
+    map_datasets = []
+
+    for obs_id in [123, 124, ...]:
+        bkg_model = BackgroundModel.read("obs-{}/background.fits".format(obs_id))
+        dataset = MapDataset(model=model, background=bkg_model)
+        map_datasets.append(dataset)
+
+    datasets = Datasets(map_datasets)
+
+    # check whether all contained datasets are of the same type
+    datasets.is_all_same_type
+
+    datasets.stat()
+
+    fit = Fit(datasets)
+
+    # or maybe simpler
+
+    fit = Fit(map_datasets)
+    fit.optimize()
+
+
+The linking of parameters of the spectral model is natively achieved, as the same
+instance of the spectral model is passed to two different datasets, while the
+background model is created for every dataset separately.
+
+A joined spectral fit across multiple observations:
+
+.. code ::
+
+    model = SpectralModel()
+
+    spectrum_dataset_1 = SpectrumDataset(model, ...)
+    spectrum_dataset_2 = SpectrumDataset(model, ...)
+
+    datasets = Datasets([spectrum_dataset_1, spectrum_dataset_2])
+
+    fit = Fit(datasets)
+    fit.optimize()
+
+
+Combined spectral / flux points analysis:
+
+.. code ::
+
+    model = SpectralModel()
+
+    spectrum_dataset = SpectrumDataset(model=model)
+
+    flux_points = FluxPoints.read()
+    flux_point_dataset = FluxPointsDataset(flux_points=flux_points, model=model, likelihood="chi2")
+
+    datasets = Datasets([flux_point_dataset, spectrum_dataset])
+
+    fit = Fit(datasets)
+    fit.optimize()
+
+
+Dataset serialization
+---------------------
+
+For convenience all dataset classes should support serialization, implemented
+via `.read()` and `.write()` methods. For now we only consider the serialization
+of the data of the datasets and not the of the model, which might always stay
+separate. As the dataset has to orchestrate the serialization of multiple objects,
+such as different kind of maps, flux-points etc. one option is to introduce the
+serialization with a YAML based index file:
+
+.. code ::
+
+    dataset = MapDataset.read("dataset.yaml")
+    dataset.write("dataset.yaml")
+
+Where the index file points to the various files needed for initialization of the
+dataset. Here is an example:
+
+.. code::
+
+    dataset:
+        type: map-dataset
```

Comment:
Suggest to remove the `-dataset` everywhere, i.e. just have `type: map` etc.
This is already under a `dataset` key.